glean_parser/translate.py

```diff
@@ -109,6 +110,35 @@ def translate_metrics(
     if util.report_validation_errors(all_objects):
         return 1
 
+    # We transform certain metrics typenames depending on context in the file.
```

Comment:
This seems close but maybe not quite the right place to put this logic.  However, certain translators (e.g. markdown) won't want this -- there it's probably better to represent things directly as in the metrics.yaml and/or payload.  Perhaps the way to do this is to put it in its own function, which is called from all of the language bindings that need the distinction between rates/counters/numerators/denominators, and skip it in the ones that don't...?  Or alternatively an additional bit of metadata on the `Outputter` objects to control this?

In any event, as a nit, I think this new logic should be in its own function to avoid bloating this one...